src/Filters/coefficients.jl

```diff
@@ -122,20 +122,43 @@ PolynomialRatio{:s, Int64}(LaurentPolynomial(3 + 2*s + s²), LaurentPolynomial(4
 
 """
 PolynomialRatio(b, a) = PolynomialRatio{:z}(b, a)
-function PolynomialRatio{:z}(b::LaurentPolynomial{T1}, a::LaurentPolynomial{T2}) where {T1,T2}
-    return PolynomialRatio{:z,typeof(one(T1)/one(T2))}(b, a)
-end
-function PolynomialRatio{:s}(b::LaurentPolynomial{T1}, a::LaurentPolynomial{T2}) where {T1,T2}
-    return PolynomialRatio{:s,promote_type(T1, T2)}(b, a)
-end
+const PolynomialRatioArgTs{T} = Union{T,Vector{T},LaurentPolynomial{T}} where {T<:Number}
+PolynomialRatio{:z}(b::PolynomialRatioArgTs{T1}, a::PolynomialRatioArgTs{T2}) where {T1<:Number,T2<:Number} =
+    PolynomialRatio{:z,typeof(one(T1) / one(T2))}(b, a)
+PolynomialRatio{:s}(b::PolynomialRatioArgTs{T1}, a::PolynomialRatioArgTs{T2}) where {T1<:Number,T2<:Number} =
+    PolynomialRatio{:s,promote_type(T1, T2)}(b, a)
+
+"""
+    _polyprep(D::Symbol, x::Union{T,Vector{T}}, ::Type) where {T<:Number}
+
+Converts `x` to polynomial form. If x is a `Number`, it has to be converted into
+a `Vector`, otherwise `LaurentPolynomial` dispatch goes into stack overflow
+trying to collect a 0-d array into a `Vector`.
+
+!!! warning
 
-# The DSP convention for Laplace domain is highest power first. The Polynomials.jl
-# convention is lowest power first.
-_polyprep(D::Symbol, x, T...) = LaurentPolynomial{T...}(reverse(x), D === :z ? -length(x)+1 : 0, D)
-PolynomialRatio{D,T}(b::Union{Number,Vector{<:Number}}, a::Union{Number,Vector{<:Number}}) where {D,T} =
-    PolynomialRatio{D,T}(_polyprep(D, b, T), _polyprep(D, a, T))
-PolynomialRatio{D}(b::Union{Number,Vector{<:Number}}, a::Union{Number,Vector{<:Number}}) where {D} =
-    PolynomialRatio{D}(_polyprep(D, b), _polyprep(D, a))
+    The DSP convention for Laplace domain is highest power first.\n
+    The Polynomials.jl convention is lowest power first.
+"""
+@inline _polyprep(D::Symbol, x::Union{T,Vector{T}}, ::Type{V}) where {T<:Number,V} =
+    LaurentPolynomial{V,D}(x isa Vector ? reverse(x) : [x], D === :z ? -length(x) + 1 : 0)
+
+function PolynomialRatio{:z,T}(b::Union{T1,Vector{<:T1}}, a::Union{T2,Vector{<:T2}}) where {T<:Number,T1<:Number,T2<:Number}
```

Comment:
```suggestion
function PolynomialRatio{:z,T}(b::Union{T1,Vector{T1}}, a::Union{T2,Vector{T2}}) where {T<:Number,T1<:Number,T2<:Number}
```
Or am I missing something?